src/code/z_skin_matrix.c

```diff
@@ -1,42 +1,692 @@
+    f32 cx;
+    f32 cy;
+    f32 cz;
+    f32 cw;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_matrix/func_800A735C.s")
+    //---COL1---
+    cx = mfB->xx;
+    cy = mfB->yx;
+    cz = mfB->zx;
+    cw = mfB->wx;
+    //--------
+    rx = mfA->xx;
+    ry = mfA->xy;
+    rz = mfA->xz;
+    rw = mfA->xw;
+    dest->xx = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
 
+    rx = mfA->yx;
+    ry = mfA->yy;
+    rz = mfA->yz;
+    rw = mfA->yw;
+    dest->yx = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    rx = mfA->zx;
+    ry = mfA->zy;
+    rz = mfA->zz;
+    rw = mfA->zw;
+    dest->zx = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    rx = mfA->wx;
+    ry = mfA->wy;
+    rz = mfA->wz;
+    rw = mfA->ww;
+    dest->wx = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    //---COL2---
+    cx = mfB->xy;
+    cy = mfB->yy;
+    cz = mfB->zy;
+    cw = mfB->wy;
+    //--------
+    rx = mfA->xx;
+    ry = mfA->xy;
+    rz = mfA->xz;
+    rw = mfA->xw;
+    dest->xy = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    rx = mfA->yx;
+    ry = mfA->yy;
+    rz = mfA->yz;
+    rw = mfA->yw;
+    dest->yy = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    rx = mfA->zx;
+    ry = mfA->zy;
+    rz = mfA->zz;
+    rw = mfA->zw;
+    dest->zy = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    rx = mfA->wx;
+    ry = mfA->wy;
+    rz = mfA->wz;
+    rw = mfA->ww;
+    dest->wy = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    //---COL3---
+    cx = mfB->xz;
+    cy = mfB->yz;
+    cz = mfB->zz;
+    cw = mfB->wz;
+    //--------
+    rx = mfA->xx;
+    ry = mfA->xy;
+    rz = mfA->xz;
+    rw = mfA->xw;
+    dest->xz = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    rx = mfA->yx;
+    ry = mfA->yy;
+    rz = mfA->yz;
+    rw = mfA->yw;
+    dest->yz = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    rx = mfA->zx;
+    ry = mfA->zy;
+    rz = mfA->zz;
+    rw = mfA->zw;
+    dest->zz = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    rx = mfA->wx;
+    ry = mfA->wy;
+    rz = mfA->wz;
+    rw = mfA->ww;
+    dest->wz = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    //---COL4---
+    cx = mfB->xw;
+    cy = mfB->yw;
+    cz = mfB->zw;
+    cw = mfB->ww;
+    //--------
+    rx = mfA->xx;
+    ry = mfA->xy;
+    rz = mfA->xz;
+    rw = mfA->xw;
+    dest->xw = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    rx = mfA->yx;
+    ry = mfA->yy;
+    rz = mfA->yz;
+    rw = mfA->yw;
+    dest->yw = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    rx = mfA->zx;
+    ry = mfA->zy;
+    rz = mfA->zz;
+    rw = mfA->zw;
+    dest->zw = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    rx = mfA->wx;
+    ry = mfA->wy;
+    rz = mfA->wz;
+    rw = mfA->ww;
+    dest->ww = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+}
+
+MtxF D_8012A4A0 = { 1.0f, 0.0f, 0.0f, 0.0f, 0.0f, 1.0f, 0.0f, 0.0f, 0.0f, 0.0f, 1.0f, 0.0f, 0.0f, 0.0f, 0.0f, 1.0f };
+
```

Comment:
if you want to keep this formatted in rows and columns you can use // clang-format off and // clang-format on to temporarily disable the formatter between the two commens